### PR TITLE
Add support to advertise provided features

### PR DESCRIPTION
**Pull request by lavoiesl**
```scheme
(provide "ruby") ; activated shadowenv (ruby)
(provide "ruby" "2.3.7") ; activated shadowenv (ruby:2.3.7)
```

image

### CODE REVIEW

File: dev.yml [hunk 1]
```diff
@@ -6,4 +6,4 @@ up:
 commands:
   build:     cargo build
   test:      cargo test
-  run-built: target/debug/shadowenv ''
+  run-built: target/debug/shadowenv
```
Comment (lavoiesl) [inline dev.yml]: Not sure why there was a `''` before, but it was causing `dev run` to always fail

Comment (burke) [reply dev.yml]: ah, the `hook` subcommand used to be the default command, so that was the `$__shadowenv_data` parameter.

File: src/output.rs [hunk 2]
```diff
@@ -34,12 +36,16 @@ pub fn handle_hook_error(err: Error, shellpid: u32, silent: bool) -> i32 {
...
-    let word = match activated {
-        true => "activated",
-        false => "deactivated",
-    };
-    eprintln!("\x1b[1;34m{} {}.\x1b[0m", word, SHADOWENV);
+pub fn print_activation(activated: bool, features: HashSet<Feature>) {
+    if activated {
+        eprint!("\x1b[1;34mactivated {}\x1b[1;34m", SHADOWENV);
+        for feature in features {
+            eprint!(" ({})", feature);
```
Comment (burke) [inline src/output.rs]: I think I'd prefer it to look like:

```
activated shadowenv (ruby:x.y.z, node:a.b.c)
```

Thoughts?

Comment (burke) [reply src/output.rs]:
```rust
        features
            .iter()
            .map(|s| format!("{}", s))
            .collect::<Vec<String>>()
            .join(", ");
```

File: src/lang.rs [hunk 3]
```diff
@@ -153,6 +153,32 @@ impl ShadowLang {
...
+            .add_value_with_name("provide", |name| {
+                Value::new_foreign_fn(name, move |ctx, args| {
+                    let value = ctx
+                        .scope()
+                        .get_constant(shadowenv_name)
+                        .expect("bug: shadowenv not defined");
+                    let shadowenv = <&Shadowenv as FromValueRef>::from_value_ref(&value)?;
+
+                    let feature = <&str as FromValueRef>::from_value_ref(&args[0])?;
+                    let version = match args.len() {
```
[the remark this answers is not in the document]
Comment (lavoiesl) [reply src/lang.rs]: Ah, yes, good catch

File: src/output.rs (hunk 2 above)
Comment (lavoiesl) [reply src/output.rs]: Yeah, that's nicer.

**Pull request merged**